### PR TITLE
SLI-248 Show analyzers loaded in Project settings

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    @Override public int getColumnCount() {
+      return 2;
+    }
+
+    @Override
+    public boolean isCellEditable(int row, int column) {
+      return false;
+    }
+
+    public Class<?> getColumnClass(int columnIndex) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      this.delegate = delegate;
+    }
+
+    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
+      return delegate.getTableCellRendererComponent(table, value, isSelected, false, row, column);
+    }
+  }
+
+  private static class Model extends AbstractTableModel {
+    private List<LoadedAnalyzer> rows = new ArrayList<>();
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: CRITICAL Make "rows" transient or serializable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    try {
+      Collection<LoadedAnalyzer> loadedAnalyzers = bindingManager.getFacade().getLoadedAnalyzers();
+      tableModel.set(loadedAnalyzers);
+    } catch(Exception e) {
+      LOGGER.error(e);
+      tableModel.set(Collections.emptyList());
+    }
+  }
+
+  private void createUIComponents() {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MAJOR Remove this unused private "createUIComponents" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    analyzerTable.setShowVerticalLines(false);
+    analyzerTable.getTableHeader().setReorderingAllowed(false);
+    analyzerTable.enableInputMethods(false);
+    analyzerTable.setDefaultRenderer(String.class, new NoFocusCellRenderer(new DefaultTableCellRenderer()));
+
+    final ToolbarDecorator decorator = ToolbarDecorator.createDecorator(analyzerTable);
+    tablePanel = decorator.createPanel();
+  }
+
+  private class NoFocusCellRenderer extends DefaultTableCellRenderer {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MAJOR Make this a "static" inner class. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)
MAJOR This class has 6 parents which is greater than 5 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    private final TableCellRenderer delegate;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: CRITICAL Make "delegate" transient or serializable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    public NoFocusCellRenderer(TableCellRenderer delegate) {
+      this.delegate = delegate;
+    }
+
+    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+import javax.swing.table.TableCellRenderer;
+import org.apache.commons.lang.StringUtils;
+import org.sonarlint.intellij.core.ProjectBindingManager;
+import org.sonarlint.intellij.util.SonarLintUtils;
+import org.sonarsource.sonarlint.core.client.api.connected.LoadedAnalyzer;
+
+public class SonarLintProjectAnalyzersPanel {
+  private static final Logger LOGGER = Logger.getInstance(SonarLintProjectAnalyzersPanel.class);
+  private final Project project;
+  private JBTable analyzerTable;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MINOR Remove the "analyzerTable" field and declare it as a local variable in the relevant methods. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1450)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+  private JPanel panel;
+  private JPanel tablePanel;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MAJOR Remove this unused "tablePanel" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
MINOR Remove the "tablePanel" field and declare it as a local variable in the relevant methods. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1450)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java [hunk 9]
```diff
@@ -25,6 +25,8 @@
 import java.awt.BorderLayout;
 import java.util.List;
 import javax.swing.JPanel;
+import javax.swing.event.ChangeEvent;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: MINOR Remove this unused import 'javax.swing.event.ChangeEvent'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java [hunk 10]
```diff
@@ -25,6 +25,8 @@
...
+import javax.swing.event.ChangeListener;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: MINOR Remove this unused import 'javax.swing.event.ChangeListener'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/core/ConnectedSonarLintFacade.java [hunk 11]
```diff
@@ -40,6 +40,7 @@
 import org.sonarsource.sonarlint.core.client.api.connected.ConnectedAnalysisConfiguration;
 import org.sonarsource.sonarlint.core.client.api.connected.ConnectedSonarLintEngine;
 import org.sonarsource.sonarlint.core.client.api.connected.LoadedAnalyzer;
+import org.sonarsource.sonarlint.core.client.api.connected.SonarAnalyzer;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/core/ConnectedSonarLintFacade.java]: MINOR Remove this unused import 'org.sonarsource.sonarlint.core.client.api.connected.SonarAnalyzer'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarlint/intellij/core/SonarLintFacade.java [hunk 12]
```diff
@@ -35,6 +35,8 @@
 import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
 import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
 import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
+import org.sonarsource.sonarlint.core.client.api.connected.LoadedAnalyzer;
+import org.sonarsource.sonarlint.core.client.api.connected.SonarAnalyzer;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/core/SonarLintFacade.java]: MINOR Remove this unused import 'org.sonarsource.sonarlint.core.client.api.connected.SonarAnalyzer'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 14 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+    analyzerTable.setShowVerticalLines(false);
+    analyzerTable.getTableHeader().setReorderingAllowed(false);
+    analyzerTable.enableInputMethods(false);
+    analyzerTable.setDefaultRenderer(String.class, new NoFocusCellRenderer(new DefaultTableCellRenderer()));
+
+    final ToolbarDecorator decorator = ToolbarDecorator.createDecorator(analyzerTable);
+    tablePanel = decorator.createPanel();
+  }
+
+  private static class NoFocusCellRenderer extends DefaultTableCellRenderer {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: MAJOR This class has 6 parents which is greater than 5 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    private final TableCellRenderer delegate;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectAnalyzersPanel.java]: CRITICAL Make "delegate" transient or serializable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1948)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java [hunk 15]
```diff
@@ -52,7 +54,13 @@ public SonarLintProjectSettingsPanel(Project project) {
     tabs.insertTab("Bind to SonarQube project", null, rootBindPane, "Configure the binding of modules to a SonarQube server", 0);
     tabs.insertTab("File Exclusions", null, exclusionsPanel.getComponent(), "Configure which files to exclude from analysis", 1);
     tabs.insertTab("Analysis properties", null, rootPropertiesPane, "Configure analysis properties", 2);
+    tabs.insertTab("Analyzers Info", null, analyzersPanel.getPanel(), "Check which analyzers are currently loaded", 3);
```
Comment (janos-ss) [inline src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: Maybe we could even drop the "Info", and name the tab just "Analyzers" ?

Comment (dbmeneses) [reply src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: That's fine too. @laurawacrenier  WDYT?

Comment (laurawacrenier) [reply src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectSettingsPanel.java]: I'm ok with both

**Pull request merged**